packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -98,4 +198,13 @@ class CodeSizeController {
   void changeRoot(TreemapNode newRoot) {
     _currentRoot.value = newRoot;
   }
+
+  void changeActiveDiffTreeType(DiffTreeType newDiffTreeType) {
+    _activeDiffTreeTypeNotifier.value = newDiffTreeType;
```

Comment:
add a todo to cache each diff tree so that it is less expensive to change between diff tree types. Then if we are pulling from the same data, we can use the cached tree instead of generating a new one